contrib/autostyle.lua

```diff
@@ -0,0 +1,116 @@
+--[[
+Autostyle
+Automatically apply a given style when an exif tag is present in the file. This tagged is checked with exiftool, in order to be able to match very exotic tags.
+I wrote this initially to be able to apply a style to compensate for Auto-DR from my Fujifilm camera
+
+AUTHOR
+Marc Cousin (cousinmarc@gmail.com)
+
+INSTALATION
+* copy this file in $CONFIGDIR/lua/ where CONFIGDIR
+is your darktable configuration directory
+* add the following line in the file $CONFIGDIR/luarc
+  require "autostyle"
+
+USAGE
+* set the exif configuration string in your lua configuration
+  mine, for instance, is AutoDynamicRange=200%=>DR200"
+  meaning that I automatically want to apply my DR200 style on all
+  images where exiftool returns '200%' for the AutoDynamicRange tag
+* if you want to be able to apply it manually to already imported
+  images, define a shortcut (lua shortcuts). As I couldn't find an event for
+  when a development is removed, so the autostyle won't be applied again, 
+  this shortcut is also helpful then
+* import your images, or use the shortcut on your already imported images
+* To determine which tag you want, list all tags with exiftool:
+  exiftool -j XE021351.RAF, and find the one you want to use
+  you can check it with 
+  > exiftool -AutoDynamicRange XE021351.RAF
+  Auto Dynamic Range              : 200%
+
+
+LICENSE
+GPLv2
+
+]]
+
+local darktable = require "darktable"
+
+
+-- Forward declare the functions
+local autostyle_apply_one_image,autostyle_apply,exiftool_attribute,capture
+
+-- Tested it with darktable 1.6.1 and darktable git from 2014-01-25
+darktable.configuration.check_version(...,{2,0,2},{2,1,0})
+
+-- Apply the style to an image, if it matches the tag condition
+function autostyle_apply_one_image (image)
+  -- We need the tag, the value and the style_name provided from the configuration string
+  local tag,value,style_name=string.match(darktable.preferences.read("autostyle","exif_tag","string"),"(%g+)%s*=%s*(%g+)%s*=>%s*(%g+)")
+
+  -- check they all exist (correct syntax)
+  assert (tag,"EXIF TAG not found in " .. darktable.preferences.read("autostyle","exif_tag","string"));
+  assert (value,"value to match not found in " .. darktable.preferences.read("autostyle","exif_tag","string"));
+  assert (style_name,"style name not found in " .. darktable.preferences.read("autostyle","exif_tag","string"));
+
```

Comment:
yes, I mean the asserts... normal users might not look at stdout and this is an error case a normal user can reach